TourMate/TourMate/Frontend/Common/ViewFactory/ViewFactory+PlanLogView.swift

```diff
@@ -0,0 +1,50 @@
+//
+//  ViewFactory+PlanLogView.swift
+//  TourMate
+//
+//  Created by Terence Ho on 15/4/22.
+//
+
+import Foundation
+import SwiftUI
+
+// MARK: PlanLogView components
+extension ViewFactory {
+
+    func getPlanVersionView<T: Plan>(planDisplayViewModel: PlanDisplayViewModel<T>, plan: T) -> some View {
+        let action = plan.versionNumber == 1 ? "created" : "updated"
+        let username = planDisplayViewModel.getPlanModifier(version: plan.versionNumber)?.name ?? "someone..."
+        return HStack {
+            Spacer()
+            Text("Plan version \(plan.versionNumber) \(action) by \(username)")
+                .bold()
+            Spacer()
+        }
+    }
+
+    // Will be empty if there are no views to display
+    @ViewBuilder
+    func getUpvotedUsersView(planUpvoteViewModel: PlanUpvoteViewModel, version: Int) -> some View {
+        let upvotedUsers = planUpvoteViewModel.getUpvotedUsersForVersion(version: version)
+
+        if !upvotedUsers.isEmpty {
+            let upvotedUsersText = upvotedUsers.map({ $0.name }).joined(separator: ", ")
+            let displayText = upvotedUsersText + " liked this version :)"
+
+            HStack {
+                Spacer()
+                Text(displayText)
+                Spacer()
+            }
+        }
+    }
+
+    func getComments(commentsViewModel: CommentsViewModel, version: Int) -> some View {
```

Comment:
Are we no longer using commentsview?

Reply:
I'm trying to see how to integrate the PlanLogView and CommentsView together. It didn't make sense at first to pass like `planDisplayViewModel` and `planUpvoteViewModel` into a CommentsView because it should be mainly for displaying comments.

Reply:
Should we put this in their own view instead of in the view factory?

Reply:
I will abstract the View part into a Component but I think I might keep the logic (like generating the Strings to display) inside the view factory